botcore/utils/_extensions.py

```diff
@@ -47,6 +46,4 @@ def on_error(name: str) -> NoReturn:
                 # If it lacks a setup function, it's not an extension.
                 continue
 
-        modules.add(module_info.name)
-
-    return frozenset(modules)
+        yield module_info.name
```

Comment:
The "extensions" extension still needs a frozenset of this. It can't use `bot.extensions` since that contains currently loaded extensions rather than all available extensions. Not sure if it's better to expose it as a constant, as was done before, or if it should be stored as an instance attribute of `BotBase`.

Reply:
Good point, updated here https://github.com/python-discord/bot-core/pull/42/commits/b8a245061689c45ebee05f27481837576a079e54

Reply:
Could've been left as a generator, but it doesn't really matter. I guess creating a frozenset directly would me more efficient anyway?